yt_dlp/extractor/bahamut.py

```diff
@@ -0,0 +1,111 @@
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    float_or_none,
+    smuggle_url,
+    unified_timestamp,
+    unsmuggle_url,
+)
+from ..utils.traversal import traverse_obj
+
+
+class BahamutIE(InfoExtractor):
+    _VALID_URL = r'https?://ani\.gamer\.com\.tw/animeVideo\.php\?sn=(?P<id>\d+)'
+
+    # see anime_player.js
+    RATING_TO_AGE_LIMIT = {
+        1: 0,
+        2: 6,
+        3: 12,
+        4: 15,
+        5: 18,
+        6: 18,  # age-gated, needs login
+    }
+
+    def _real_extract(self, url):
+        url, unsmuggled_data = unsmuggle_url(url, {})
+        video_id = self._match_id(url)
+        device_id = (
+            self._configuration_arg('device_id', [None], casesense=True)[0]
+            or unsmuggled_data.get('device_id')
+            or self._download_json(
+                'https://ani.gamer.com.tw/ajax/getdeviceid.php', video_id,
+                'Downloading device ID', 'Failed to download device ID',
+                impersonate=True, headers=self.geo_verification_headers())['deviceid'])
+
+        # TODO: extract metadata from webpage
+        metadata = {}
+        if api_result := self._download_json(
+                'https://api.gamer.com.tw/anime/v1/video.php', video_id,
+                'Downloading video info', 'Failed to download video info',
+                impersonate=True, query={'videoSn': video_id}).get('data'):
+
+            metadata.update(traverse_obj(api_result, ('anime', {
+                'description': 'content',
+                'thumbnail': 'cover',
+                'tags': 'tags',
+                'creators': ('director', {lambda x: [x]}),
+                'title': 'title',
+            })))
+            playlist_id = traverse_obj(api_result, ('video', 'animeSn')) or ''
+            if self._yes_playlist(playlist_id, video_id) and unsmuggled_data.get('extract_playlist') is not False:
+                return self.playlist_result(
+                    (self.url_result(
+                        # it may be better to use self.cache for storing device_id
```

Comment:
it would be better just to cache to a class variable, even

Reply:
see the wrestleuniverse extractor for an example of caching device ID that is input via extractor-arg to class var and to disk

Reply:
the device id lives for ~5-7hrs. Is it long enough to cache to disk?

Reply:
what is the consequence of using a new device id every time?

Reply:
The site's js retrieves a new device id every time, and I never meet with an API rate limit. I just wanted to reduce the requests made